MiniProject 1/VideoViewController.swift

```diff
@@ -11,43 +11,115 @@ import AVFoundation
 
 class VideoViewController: UIViewController
 {
-    @IBOutlet weak var playerView: AVPlayerView!
     private var playerItem: AVPlayerItem?
+    {
+        willSet
+        {
+            NSNotificationCenter.defaultCenter().removeObserver(self,
+            name: AVPlayerItemDidPlayToEndTimeNotification,
+            object: videoPlayer?.currentItem)
+        }
+        didSet
+        {
+            NSNotificationCenter.defaultCenter().addObserver(self,
+            selector: "restartVideo",
+            name: AVPlayerItemDidPlayToEndTimeNotification,
+            object: videoPlayer?.currentItem)
+        }
+    }
+    
     private var videoPlayer: AVPlayer?
     private var videoNumber = 0
+    private var movingCount: CGFloat = 0
     private var urls = ["http://fun.siz.io/stories/142893791787803c7fb48f4d/0.mp4",
                 "http://fun.siz.io/stories/1429024267698bb127fbd1bf/0.mp4",
                 "http://fun.siz.io/stories/1429021517853eac7ce1fc29/0.mp4",
                 "http://fun.siz.io/stories/1429018976114f45db3a5f88/0.mp4"]
     
-    @IBAction func mainButtons(sender: UIButton)
+    @IBOutlet weak var playerView: AVPlayerView!
+    {
+        didSet
+        {
+            playerView.addGestureRecognizer(UIPanGestureRecognizer(target:self, action: "changeVideo:"))
+        }
+    }
+
+    @IBAction func previousVideo()
+    {
+        if(videoNumber > 0)
+        {
+            videoNumber--
+        }
+        else if(videoNumber == 0)
+        {
+            videoNumber = urls.count-1
+        }
+        
+        playVideo()
+    }
+    
+    @IBAction func nextVideo()
     {
-        let buttonName = sender.currentTitle!
+        if(videoNumber < urls.count-1)
+        {
+            videoNumber++
+        }
+        else if(videoNumber == urls.count-1)
+        {
+            videoNumber = 0
+        }
         
-        switch buttonName
+        playVideo()
+    }
+    
+    @IBAction func changeVideo(gesture: UIPanGestureRecognizer)
+    {
+        let translation = gesture.translationInView(view)
+        
+        switch gesture.state
         {
-        case"PREV":
-            if(videoNumber > 0)
+        case .Changed:
+            if let view = gesture.view
             {
-                videoNumber--
+                println("\(translation.x)")
+                println("\(movingCount)")
```

Comment:
You can leave some debug information but this debug need to be clear when I read logs